rs/sns/governance/src/governance.rs

```diff
@@ -2529,13 +2545,45 @@ impl Governance {
             })
     }
 
+    pub fn check_no_other_upgrades_in_progress(
+        &self,
+        proposal_id: u64,
+    ) -> Result<(), GovernanceError> {
+        let upgrade_proposals_in_progress = self.upgrade_proposals_in_progress();
```

Comment:
Let's add a TODO or Jira ticket to add another failure condition:

If the `upgrade_proposals_in_progress` does not contain `proposal_id`, then this would indicate corrupted canister state.

Reply:
Implemented!